### DISCUSSION TITLE
lasio sprint - 't21-hack-lasio'

### DISCUSSION THREAD

**Discussion by kinverarity1** [softwareunderground/transform-2021-hackathon#17] [Projects]
## Slack channel
[t21-hack-lasio](https://swung.slack.com/archives/C01TMSG9TFV/p1617862924000300)

## People
* @kinverarity1
* @dcslagel

## Description
This hackathon aims to fix some long-standing bugs with lasio, and also implement a range of improvements ranging from simple to complex ones. Any help is welcome, including newcomers.

## Goals
* Fix some of the existing bugs: https://github.com/kinverarity1/lasio/issues?q=is%3Aopen+is%3Aissue+label%3Abug
* Support for LAS 3 features (multiple data sections, format and association tags)
* Support for non-numeric data in the data section
* Faster reading of LAS files (starting with benchmarks for the current slug-like speed ;-))
* Improve flexibility and reliability for how LAS files are written out by lasio (and improve the test suite in this regard)
* Improve the Sphinx documentation, which is missing detail for many features or aspects of how the package works.
* Look for places to interact more smoothly with [welly](https://github.com/agile-geoscience/welly)

## Resources
* [Transform 21 lasio project board](https://github.com/kinverarity1/lasio/projects/2) - this is where we will coordinate issues and who is working on what.
* Guide to [contributing to lasio](https://lasio.readthedocs.io/en/latest/contributing.html)
* Software underground Slack channel #t21-hack-lasio (see [link here](https://swung.slack.com/archives/C01TMSG9TFV/p1617862924000300))

## Skills needed (if any)

It will be helpful if you are familiar with Python and GitHub, but newcomers are welcome and it will be a good opportunity for you to learn!

**Comment by dcslagel** [comment]
#### Information on contributing:
https://lasio.readthedocs.io/en/latest/contributing.html

**Comment by filippo82** [comment]
[T21 Hackathon Idea Mashup](https://softwareunderground.org/events/2021/4/7/t21-hackathon-idea-mashup) today (or tomorrow depending on your time zone) Wednesday April 14th at 16:00 UTC. Come to Slack to find the link for the meeting!
